File: 2681-put-marbles-in-bags/put-marbles-in-bags.java
class Solution {
    public long putMarbles(int[] weights, int k) {
        int n = weights.length;
        if (k == 1 || n == k) return 0;
        int[] costs = new int[n-1];
        for (int i = 0; i < n - 1; ++i) {
            costs[i] = weights[i] + weights[i + 1];
        }
        Arrays.sort(costs);
        long res = 0l;
        for (int i = 0; i < k - 1; ++i) {
            res += costs[n - 2 - i] - costs[i];
        }
        return res;
    }
}

File: 2681-put-marbles-in-bags/put-marbles-in-bags.py
class Solution:
    def putMarbles(self, weights: List[int], k: int) -> int:
        n = len(weights)
        if k == 1 or k == n:
            return 0
        for i in range(n-1):
            weights[i] = weights[i] + weights[i+1]
        weights.pop()
        weights.sort()
        return sum(weights[-k+1:]) - sum(weights[:k-1])